crates/rspack_plugin_javascript/src/dependency/commonjs/common_js_exports_dependency.rs

```diff
@@ -0,0 +1,240 @@
+      runtime_requirements,
+      ..
+    } = code_generatable_context;
+
+    let mgm = compilation
+      .module_graph
+      .module_graph_module_by_identifier(&module.identifier())
+      .expect("should have mgm");
+
+    let used = compilation
+      .module_graph
+      .get_exports_info(&module.identifier())
+      .id
+      .get_used_name(&compilation.module_graph, *runtime, self.names.clone());
+
+    let exports_argument = mgm.get_exports_argument();
+    let module_argument = mgm.get_module_argument();
+
+    let base = if self.base.is_exports() {
+      runtime_requirements.insert(RuntimeGlobals::EXPORTS);
+      exports_argument.to_string()
+    } else if self.base.is_module_exports() {
+      runtime_requirements.insert(RuntimeGlobals::MODULE);
+      format!("{}.exports", module_argument)
+    } else if self.base.is_this() {
+      runtime_requirements.insert(RuntimeGlobals::THIS_AS_EXPORTS);
+      "this".to_string()
+    } else {
+      panic!("Unexpected base type");
+    };
+
+    if self.base.is_expression() {
+      if let Some(used) = used {
+        source.replace(
+          self.range.0,
+          self.range.1,
+          &format!(
+            "{}{}",
+            base,
+            property_access(
+              match used {
+                UsedName::Str(name) => vec![name].into_iter(),
+                UsedName::Vec(names) => names.into_iter(),
+              },
+              0
+            )
+          ),
+          None,
+        )
+      } else {
+        init_fragments.push(
+          NormalInitFragment::new(
+            "var __webpack_unused_export__;\n".to_string(),
+            InitFragmentStage::StageConstants,
+            0,
+            InitFragmentKey::uniqie(),
+            None,
+          )
+          .boxed(),
+        );
+        source.replace(
+          self.range.0,
+          self.range.1,
+          "__webpack_unused_export__",
+          None,
+        );
+      }
+    } else if self.base.is_define_property() {
+      if let Some(value_range) = self.value_range {
+        if let Some(used) = used {
+          if let UsedName::Vec(used) = used {
+            source.replace(
+              self.range.0,
+              value_range.0,
+              &format!(
+                "Object.defineProperty({}{}, {}, (",
+                base,
+                property_access(used[0..used.len() - 1].iter(), 0),
+                serde_json::to_string(&used.last())
+                  .expect("Unexpected render define property base")
+              ),
+              None,
+            );
+            source.replace(value_range.1, self.range.1, "))", None);
+          } else {
+            panic!("Unexpected base type");
+          }
+        } else {
+          init_fragments.push(
+            NormalInitFragment::new(
+              "var __webpack_unused_export__;\n".to_string(),
+              InitFragmentStage::StageConstants,
+              0,
+              InitFragmentKey::uniqie(),
```

Comment:
Do you mean `unique`?

Reply:
Yes, but the previous code was unique. I'll modify it.